### DISCUSSION TITLE
Having a difficult time downloading Large files, and missing files.

### DISCUSSION THREAD

**Discussion by CarotMagic** [RhetTbull/osxphotos#1305] [Q&A]
Hello again. Downloading photos has been easy and great, however, there seems to be some files left out of exporting to my folder. 

**This is the command I use**

osxphotos export \
--album "L & A" \
--exiftool \
--exiftool-option -api \
--exiftool-option largefilesupport=1 \
--use-photokit \
--from-date 2023-08-15 \
--to-date 2023-09-15 \
--update \
--touch-file \
--download-missing \
--keyword-template {album} \
--directory "{folder_album}" \
--exportdb \
--db /Users/landy/Pictures/PhotosLibrary.photoslibrary \
"/Volumes/T7 Touch/iPhone Backup"

**And this is the results I get from running that command**

WARNING: photos_library is deprecated. The PHOTOS_LIBRARY argument is deprecated and will be removed in a future version of osxphotos. Use --library instead to specify the Photos Library path.
Warning: export database is '--db' but found '.osxphotos_export.db' in 
/Volumes/T7 Touch/iPhone Backup; using '--db'
Warning: export database '/Users/landy/--db' is in a different directory than 
export destination '/Volumes/T7 Touch/iPhone Backup'
Exporting 47 photos to /Volumes/T7 Touch/iPhone Backup...
Exporting 47 photos ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Processed: 47 photos, exported: 0, updated: 0, skipped: 85, updated EXIF data: 0, missing: 13, error: 0, touched date: 0
Elapsed time: 0:00:00
landy@Landers-MacBook-Pro ~ % 

Why is it not downloading the missing files? I search through the downloaded folder to see which files is was missing, and would compare it to my photos app, and the files are there!

**Comment by RhetTbull** [comment]
First, you should heed the warnings:

```
Warning: export database is '--db' but found '.osxphotos_export.db' in
/Volumes/T7 Touch/iPhone Backup; using '--db'
Warning: export database '/Users/landy/--db' is in a different directory than
export destination '/Volumes/T7 Touch/iPhone Backup'
```

Why are you specifying `--exportdb`? It must be passed a path to the export database. This is an advanced feature you likely don't need. OSXPhotos will create the export database in the export folder. You didn't pass an export database and so osxphotos used the next argument `--db` as the database. 

I don't know why the files are missing nor why osxphotos isn't able to download them.  Remove the `--exportdb` argument and add `--use-photokit` to see if that downloads them.  You should also check `Settings > Privacy & Security > Automation` and look under `Terminal` and verify that `Photos` is enabled so that the terminal can script Photos (to tell it to download the photos). You should have been been prompted to do this but if you didn't grant permission, osxphotos can't download the missing files.

If the files are very large, it's also possible they're timing out. But start with steps above and we'll go from there.

**Reply by CarotMagic** [reply]
I have removed --exportdb and the warnings went away.

--use-photokit was already on the commands

All settings have been checked and is enabled.

It still will not download the missing files.

**Reply by RhetTbull** [reply]
Add `--verbose` and re-run. Then note the UUID (unique identifier) of a missing photo that is printed out. Then run the following and send me the output:

`osxphotos debug-dump --dump photos --uuid E9BC5C36-7CD1-40A1-A72B-8B8FAC227D51 > debug.log`

In the command above, replace `E9BC5C36-7CD1-40A1-A72B-8B8FAC227D51 ` with the UUID of one of your missing photos.

**Reply by RhetTbull** [reply]
If you don't want to post the debug output here, you can send to me at rturnbull+git@gmail.com

**Reply by CarotMagic** [reply]
I have emailed you the debug log.
